haystack_experimental/components/tools/openai/function_caller.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import List
+
+from haystack import component
+from haystack.dataclasses import ChatMessage
+
+
+@component
+class OpenAIFunctionCaller:
```

Comment:
I'd also add a few unit tests (follow the pattern used in other components) and an integration test.

Reply:
done